receiver/prometheusreceiver/internal/transaction.go

```diff
@@ -139,6 +139,27 @@ func (t *transaction) Append(ref storage.SeriesRef, ls labels.Labels, atMs int64
 }
 
 func (t *transaction) AppendExemplar(ref storage.SeriesRef, l labels.Labels, e exemplar.Exemplar) (storage.SeriesRef, error) {
+	select {
+	case <-t.ctx.Done():
+		return 0, errTransactionAborted
+	default:
+	}
+
+	if dupLabel, hasDup := l.HasDuplicateLabelNames(); hasDup {
+		return 0, fmt.Errorf("invalid sample: non-unique label names: %q", dupLabel)
+	}
+
+	mn := l.Get(model.MetricNameLabel)
+	if mn == "" {
+		return 0, errMetricNameNotFound
+	}
+	fn := normalizeMetricName(mn)
+	if mf, ok := t.families[fn]; ok && mf != nil {
+		err := mf.addExemplar(l, e)
+		if err != nil {
+			return 0, err
+		}
+	}
```

Comment:
This is partially duplicated with Append. Can we extract the `getOrCreate` logic? Also if the first thing we receive about a metric family is an exemplar, do we want to not create the family and drop the exemplar?

Reply:
Done, added a new funciton `getOrCreateMetricFamily` and used this function in `Append` and `AppendExemplar` methods